conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -0,0 +1,111 @@
+from conan.tools._compilers import architecture_flag, build_type_flags
+from conan.tools.env import Environment
+# FIXME: need to refactor this import and bring to conan.tools
+from conans.client.build.cppstd_flags import cppstd_flag_new
+
+
+class AutotoolsToolchain:
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        build_type = self._conanfile.settings.get_safe("build_type")
+
+        # TODO: compiler.runtime for Visual studio?
+        # defines
+        self.ndebug = None
+        if build_type in ['Release', 'RelWithDebInfo', 'MinSizeRel']:
+            self.ndebug = "NDEBUG"
+        self.gcc_cxx11_abi = self._cxx11_abi_define()
+        self.defines = []
+
+        # cxxflags, cflags
+        self.cxxflags = []
+        self.cflags = []
+        self.ldflags = []
+        self.libcxx = self._libcxx()
+        self.fpic = self._conanfile.options.get_safe("fPIC")
+
+        # FIXME: This needs to be imported here into conan.tools
+        self.cppstd = cppstd_flag_new(self._conanfile.settings)
+        self.arch_flag = architecture_flag(self._conanfile.settings)
+        # TODO: This is also covering compilers like Visual Studio, necessary to test it (&remove?)
+        self.build_type_flags = build_type_flags(self._conanfile.settings)
+
+    def _rpaths_link(self):
+        # TODO: Not implemented yet
+        pass
+
+    # TODO: Apple: tools.apple_deployment_target_flag,
+    # TODO:  tools.XCRun(self._conanfile.settings).sdk_path
+    # TODO: "-arch", tools.to_apple_arch(self._arch)
+
+    def _cxx11_abi_define(self):
```

Comment:
belongs to `conan/tools/_compilers.py`?

Reply:
The idea is to factor out to a common place like ``_compilers.py`` only when there is a repetition/duplicated code needed somewhere else.